content/lib/jquery-ui-slider-pips.js

```diff
@@ -0,0 +1,696 @@
+                // "", string
+
+                step: ( pips > 100 ) ? Math.floor( pips * 0.05 ) : 1,
+                // number
+
+                formatLabel: function(value) {
+                    return this.prefix + value + this.suffix;
+                }
+                // function
+                // must return a value to display in the pip labels
+
+            };
+
+            $.extend( options, settings );
+
+            slider.options.pipStep = options.step;
+
+            // get rid of all pips that might already exist.
+            slider.element
+                .addClass("ui-slider-pips")
+                .find(".ui-slider-pip")
+                .remove();
+
+            // small object with functions for marking pips as selected.
+
+            var selectPip = {
+
+                single: function(value) {
+
+                    this.resetClasses();
+
+                    $pips
+                        .filter(".ui-slider-pip-" + this.classLabel(value) )
+                        .addClass("ui-slider-pip-selected");
+
+                    if ( slider.options.range ) {
+
+                        $pips.each(function(k,v) {
+
+                            var pipVal = $(v).children(".ui-slider-label").data("value");
+
+                            if (( slider.options.range === "min" && pipVal < value ) ||
+                                ( slider.options.range === "max" && pipVal > value )) {
+
+                                $(v).addClass("ui-slider-pip-inrange");
+
+                            }
+
+                        });
+
+                    }
+
+                },
+
+                range: function(values) {
+
+                    this.resetClasses();
+
+                    for( i = 0; i < values.length; i++ ) {
+
+                        $pips
+                            .filter(".ui-slider-pip-" + this.classLabel(values[i]) )
+                            .addClass("ui-slider-pip-selected-" + (i+1) );
+
+                    }
+
+                    if ( slider.options.range ) {
+
+                        $pips.each(function(k,v) {
+
+                            var pipVal = $(v).children(".ui-slider-label").data("value");
+
+                            if( pipVal > values[0] && pipVal < values[1] ) {
+
+                                $(v).addClass("ui-slider-pip-inrange");
+
+                            }
+
+                        });
+
+                    }
+
+                },
+
+                classLabel: function(value) {
+
+                    return value.toString().replace(".","-");
+
+                },
+
+                resetClasses: function() {
+
+                    var regex = /(^|\s*)(ui-slider-pip-selected|ui-slider-pip-inrange)(-{1,2}\d+|\s|$)/gi;
+
+                    $pips.removeClass( function (index, css) {
+                        return ( css.match(regex) || [] ).join(" ");
+                    });
+
+                }
+
+            };
+
+            function getClosestHandle( val ) {
+
```

Comment:
Hey Yana, thanks for commenting! This isn't my code, it's part of the jquery-ui-slider-pips library that I use to help create the slider. You have the possibility of multiple handles on a slider with this library (I only used one handle). To clarify, the handle is the thing that you drag around on the slider.